createdisk.sh

```diff
@@ -94,12 +94,38 @@ if podman manifest inspect quay.io/crcont/routes-controller:${OPENSHIFT_VERSION}
     image_tag=${OPENSHIFT_VERSION}
 fi
 
+# create the tap device interface with specified mac address
+# this mac address is used to allocate a specific IP to the VM
+# when tap device is in use.
+${SSH} core@${VM_IP} 'sudo bash -x -s' <<EOF
+  nmcli connection add type tun ifname tap0 con-name tap0 mode tap autoconnect yes 802-3-ethernet.cloned-mac-address 5A:94:EF:E4:0C:EE
+EOF
+
+
 # Add gvisor-tap-vsock service
 ${SSH} core@${VM_IP} 'sudo bash -x -s' <<EOF
-  podman create --name=gvisor-tap-vsock --privileged --net=host -v /etc/resolv.conf:/etc/resolv.conf -it quay.io/crcont/gvisor-tap-vsock:latest
-  podman generate systemd --restart-policy=no gvisor-tap-vsock > /etc/systemd/system/gvisor-tap-vsock.service
+  podman create --name=gvisor-tap-vsock quay.io/crcont/gvisor-tap-vsock:latest
+  podman cp gvisor-tap-vsock:/vm /usr/local/bin/gvforwarder
+  podman rm gvisor-tap-vsock
+  tee /etc/systemd/system/gv-user-network@.service <<TEE
+[Unit]
+Description=gvisor-tap-vsock Network Traffic Forwarder
+After=NetworkManager.service
```

Comment:
Should this be `Before=NetworkManager.service` ? otherwise it starts very late after boot

Reply:
@anjannath networkManager is responsible to activate tun/tap device so it that is not even active how would this work.

Reply:
> Should this be `Before=NetworkManager.service` ? otherwise it starts very late after boot

Is it `NetworkManager.service` which would start late? or `network-online`?
Since we have `After=sys-devices-virtual-net-%i.device` , do we even need the reference to `NetworkManager.service`?

Reply:
I spoke with Anjan, and the TAP creation is independent of the `network` which configures. But the current, `After` is too undetermined, as it can start at any point after; anything with priority or dependency goes first.

Reply:
In short, rather use `Before` than `After` statements, as that ensures the order.